File: evidence/generated-systems/golang/analytics-on-golang/analytics-on-golang/internal/model/AnalyticsWorkspace.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// AnalyticsWorkspace Declaration
//==============================================================
type AnalyticsWorkspace struct {
    gorm.Model
     Name                                    string
    BusinessDomain                                    string
    OwnerTeam                                    string
     Datasets           []DataSet `gorm:"foreignKey:DatasetsFromAnalyticsWorkspaceId"`
     DataSources           []DataSource `gorm:"foreignKey:DataSourcesFromAnalyticsWorkspaceId"`
     Pipelines           []DataPipeline `gorm:"foreignKey:PipelinesFromAnalyticsWorkspaceId"`
     Dashboards           []Dashboard `gorm:"foreignKey:DashboardsFromAnalyticsWorkspaceId"`
     Reports           []Report `gorm:"foreignKey:ReportsFromAnalyticsWorkspaceId"`
     Notebooks           []Notebook `gorm:"foreignKey:NotebooksFromAnalyticsWorkspaceId"`
     Models           []Model_ `gorm:"foreignKey:ModelsFromAnalyticsWorkspaceId"`
     FeatureSets           []FeatureSet `gorm:"foreignKey:FeatureSetsFromAnalyticsWorkspaceId"`
     Policies           []AccessPolicy `gorm:"foreignKey:PoliciesFromAnalyticsWorkspaceId"`
     LineageNodes           []LineageNode `gorm:"foreignKey:LineageNodesFromAnalyticsWorkspaceId"`
    GovernanceTier                      GovernanceTier

// parent associations as their child

}

File: evidence/generated-systems/angular/analytics-on-angular/src/app/models/AnalyticsWorkspace.js

// Define collection and schema for AnalyticsWorkspace
export interface AnalyticsWorkspace {
    name:
	type : string
    businessDomain:
	type : string
    ownerTeam:
	type : string
    Datasets:
 	type : [{ type: Schema.Types.ObjectId, ref: 'DataSet' }]
    DataSources:
 	type : [{ type: Schema.Types.ObjectId, ref: 'DataSource' }]
    Pipelines:
 	type : [{ type: Schema.Types.ObjectId, ref: 'DataPipeline' }]
    Dashboards:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Dashboard' }]
    Reports:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Report' }]
    Notebooks:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Notebook' }]
    Models:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Model_' }]
    FeatureSets:
 	type : [{ type: Schema.Types.ObjectId, ref: 'FeatureSet' }]
    Policies:
 	type : [{ type: Schema.Types.ObjectId, ref: 'AccessPolicy' }]
    LineageNodes:
 	type : [{ type: Schema.Types.ObjectId, ref: 'LineageNode' }]
    GovernanceTier:
 	type : String
#
    collection: 'analyticsWorkspaces'
}
